Also, looks like unit tests are failing:
```py
>       self.assertEqual(gym_info["Baierl Rec Center"], "200")
E       AssertionError: '0' != '200'
E       - 0
E       + 200
```
You can find more details in the test logs

--- a/pittapi/gym.py
+++ b/pittapi/gym.py
@@ -59,7 +59,7 @@ def get_all_gyms_info() -> List[Gym]:
     return gyms
 
 
-GYM_Names = [
+GYM_NAMES = [
     "Baierl Rec Center",
     "Bellefield Hall: Fitness Center & Weight Room",
     "Bellefield Hall: Court & Dance Studio",
@@ -74,7 +74,7 @@ GYM_Names = [
 def get_gym_information(gym_name: str) -> Gym:
     """Fetches the information of a singular gym as a tuple"""
     info = get_all_gyms_info()
-    if gym_name in GYM_Names:
+    if gym_name in GYM_NAMES:
         for gym in info:
             if gym.name == gym_name:
                 return gym

--- a/pittapi/gym.py
+++ b/pittapi/gym.py
@@ -19,14 +19,21 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 
 from bs4 import BeautifulSoup
 import requests
-
+from typing import NamedTuple, List
 
 GYM_URL = "https://connect2concepts.com/connect2/?type=bar&key=17c2cbcb-ec92-4178-a5f5-c4860330aea0"
 
 
-def fetch_gym_info() -> dict:
-    """Fetches dictionary of Live Facility Counts of all gyms"""
-    gym_dict = {}
+class Gym(NamedTuple):
+    name: str
+    date: str
+    count: int
+    percentage: int
+
+
+def get_all_gyms_info() -> List[Gym]:
+    """Fetches list of Gym named tuples with all gym information"""
+    gyms = []
     # Was getting a Mod Security Error
     # Fix: https://stackoverflow.com/questions/61968521/python-web-scraping-request-errormod-security
     headers = {
@@ -41,45 +48,35 @@ def fetch_gym_info() -> dict:
     for gym in gym_info_list:
         text = gym.get_text("|", strip=True)
         info = text.split("|")
-        gym_dict[info[0]] = info[2][12:]
-    return gym_dict
+        name = info[0]
+        count = int(info[2][12:])
+        date = info[3][9:]
+        try:
+            percentage = int(info[4].rstrip("%"))
+        except ValueError:
+            percentage = 0
+        gyms.append(Gym(name=name, date=date, count=count, percentage=percentage))
+    return gyms
 
 
-def get_baierl_rec_count():
-    info = fetch_gym_info()
-    return info["Baierl Rec Center"]
+GYM_Names = [
+    "Baierl Rec Center",
+    "Bellefield Hall: Fitness Center & Weight Room",
+    "Bellefield Hall: Court & Dance Studio",
+    "Trees Hall: Fitness Center",
+    "Trees Hall: Courts",
+    "Trees Hall: Racquetball Courts & Multipurpose Room",
+    "William Pitt Union",
+    "Pitt Sports Dome",
+]
 
 
-def get_bellfield_fitness_count():
-    info = fetch_gym_info()
-    return info["Bellefield Hall: Fitness Center & Weight Room"]
-
-
-def get_bellfield_court_count():
-    info = fetch_gym_info()
-    return info["Bellefield Hall: Court & Dance Studio"]
-
-
-def get_trees_fitness_count():
-    info = fetch_gym_info()
-    return info["Trees Hall: Fitness Center"]
-
-
-def get_trees_court_count():
-    info = fetch_gym_info()
-    return info["Trees Hall: Courts"]
-
-
-def get_trees_raquetball_court_count():
-    info = fetch_gym_info()
-    return info["Trees Hall: Racquetball Courts & Multipurpose Room"]
-
-
-def get_willaim_pitt_fittness_count():
-    info = fetch_gym_info()
-    return info["William Pitt Union"]
-
-
-def get_sports_dome_count():
-    info = fetch_gym_info()
-    return info["Pitt Sports Dome"]
+def get_gym_information(gym_name: str) -> Gym:
+    """Fetches the information of a singular gym as a tuple"""
+    info = get_all_gyms_info()
+    if gym_name in GYM_Names:
+        for gym in info:
+            if gym.name == gym_name:
+                return gym
+    else:
+        return None

--- a/tests/gym_test.py
+++ b/tests/gym_test.py
@@ -1,8 +1,7 @@
 import unittest
 import responses
 from pittapi import gym
-
-GYM_URL = "https://connect2concepts.com/connect2/?type=bar&key=17c2cbcb-ec92-4178-a5f5-c4860330aea0"
+from tests.mocks.gym_mocks import mock_gym_html
 
 
 class GymTest(unittest.TestCase):
@@ -11,26 +10,32 @@ class GymTest(unittest.TestCase):
 
     @responses.activate
     def test_fetch_gym_info(self):
-        # Mock the HTML response
-        mock_html = """
-        <div class="barChart">Baierl Rec Center|Current: 100|Capacity: 200</div>
-        <div class="barChart">Bellefield Hall: Fitness Center & Weight Room|Current: 50|Capacity: 150</div>
-        <div class="barChart">Bellefield Hall: Court & Dance Studio|Current: 30|Capacity: 80</div>
-        <div class="barChart">Trees Hall: Fitness Center|Current: 70|Capacity: 120</div>
-        <div class="barChart">Trees Hall: Courts|Current: 20|Capacity: 60</div>
-        <div class="barChart">Trees Hall: Racquetball Courts & Multipurpose Room|Current: 10|Capacity: 40</div>
-        <div class="barChart">William Pitt Union|Current: 25|Capacity: 100</div>
-        <div class="barChart">Pitt Sports Dome|Current: 15|Capacity: 75</div>
-        """
 
-        responses.add(responses.GET, GYM_URL, body=mock_html, status=200)
+        responses.add(responses.GET, gym.GYM_URL, body=mock_gym_html, status=200)
 
-        gym_info = gym.fetch_gym_info()
-        self.assertEqual(gym_info["Baierl Rec Center"], "200")
-        self.assertEqual(gym_info["Bellefield Hall: Fitness Center & Weight Room"], "150")
-        self.assertEqual(gym_info["Bellefield Hall: Court & Dance Studio"], "80")
-        self.assertEqual(gym_info["Trees Hall: Fitness Center"], "120")
-        self.assertEqual(gym_info["Trees Hall: Courts"], "60")
-        self.assertEqual(gym_info["Trees Hall: Racquetball Courts & Multipurpose Room"], "40")
-        self.assertEqual(gym_info["William Pitt Union"], "100")
-        self.assertEqual(gym_info["Pitt Sports Dome"], "75")
+        gym_info = gym.get_all_gyms_info()
+        expected_info = [
+            gym.Gym(name="Baierl Rec Center", date="07/09/2024 09:05 AM", count=100, percentage=50),
+            gym.Gym(name="Bellefield Hall: Fitness Center & Weight Room", date="07/09/2024 09:05 AM", count=50, percentage=33),
+            gym.Gym(name="Bellefield Hall: Court & Dance Studio", date="07/09/2024 09:05 AM", count=30, percentage=38),
+            gym.Gym(name="Trees Hall: Fitness Center", date="07/09/2024 09:05 AM", count=70, percentage=58),
+            gym.Gym(name="Trees Hall: Courts", date="07/09/2024 09:05 AM", count=20, percentage=33),
+            gym.Gym(
+                name="Trees Hall: Racquetball Courts & Multipurpose Room",
+                date="07/09/2024 09:05 AM",
+                count=10,
+                percentage=25,
+            ),
+            gym.Gym(name="William Pitt Union", date="07/09/2024 09:05 AM", count=25, percentage=25),
+            gym.Gym(name="Pitt Sports Dome", date="07/09/2024 09:05 AM", count=15, percentage=20),
+        ]
+
+        self.assertEqual(gym_info, expected_info)
+
+    @responses.activate
+    def test_get_gym_information(self):
+        responses.add(responses.GET, gym.GYM_URL, body=mock_gym_html, status=200)
+
+        gym_info = gym.get_gym_information("Baierl Rec Center")
+        expected_info = gym.Gym(name="Baierl Rec Center", date="07/09/2024 09:05 AM", count=100, percentage=50)
+        self.assertEqual(gym_info, expected_info)

--- a/tests/mocks/gym_mocks.py
+++ b/tests/mocks/gym_mocks.py
@@ -1,0 +1,122 @@
+mock_gym_html = """
+        <div class="barChart" style="width:90%;">
+            Baierl Rec Center
+            <br>
+            <span style="color:green">(Open)</span>
+            <br>
+            Last Count: 100
+            <br>
+            Updated: 07/09/2024 09:05 AM
+            <div class="barChart_row" data-value="50.0">
+            <span class="barChart__value">50%</span>
+            <span class="barChart__bar" style=" background: rgb(239, 239, 239);">
+            <span class="barChart__barFill" style="background: #2A5B84;"></span>
+            </span>
+            </div>
+        </div>
+        <div class="barChart" style="width:90%;">
+            Bellefield Hall: Fitness Center & Weight Room
+            <br>
+            <span style="color:green">(Open)</span>
+            <br>
+            Last Count: 50
+            <br>
+            Updated: 07/09/2024 09:05 AM
+            <div class="barChart_row" data-value="33.0">
+            <span class="barChart__value">33%</span>
+            <span class="barChart__bar" style=" background: rgb(239, 239, 239);">
+            <span class="barChart__barFill" style="background: #2A5B84;"></span>
+            </span>
+            </div>
+        </div>
+        <div class="barChart" style="width:90%;">
+            Bellefield Hall: Court & Dance Studio
+            <br>
+            <span style="color:green">(Open)</span>
+            <br>
+            Last Count: 30
+            <br>
+            Updated: 07/09/2024 09:05 AM
+            <div class="barChart_row" data-value="38.0">
+            <span class="barChart__value">38%</span>
+            <span class="barChart__bar" style=" background: rgb(239, 239, 239);">
+            <span class="barChart__barFill" style="background: #2A5B84;"></span>
+            </span>
+            </div>
+        </div>
+        <div class="barChart" style="width:90%;">
+            Trees Hall: Fitness Center
+            <br>
+            <span style="color:green">(Open)</span>
+            <br>
+            Last Count: 70
+            <br>
+            Updated: 07/09/2024 09:05 AM
+            <div class="barChart_row" data-value="58.0">
+            <span class="barChart__value">58%</span>
+            <span class="barChart__bar" style=" background: rgb(239, 239, 239);">
+            <span class="barChart__barFill" style="background: #2A5B84;"></span>
+            </span>
+            </div>
+        </div>
+        <div class="barChart" style="width:90%;">
+            Trees Hall: Courts
+            <br>
+            <span style="color:green">(Open)</span>
+            <br>
+            Last Count: 20
+            <br>
+            Updated: 07/09/2024 09:05 AM
+            <div class="barChart_row" data-value="33.0">
+            <span class="barChart__value">33%</span>
+            <span class="barChart__bar" style=" background: rgb(239, 239, 239);">
+            <span class="barChart__barFill" style="background: #2A5B84;"></span>
+            </span>
+            </div>
+        </div>
+        <div class="barChart" style="width:90%;">
+            Trees Hall: Racquetball Courts & Multipurpose Room
+            <br>
+            <span style="color:green">(Open)</span>
+            <br>
+            Last Count: 10
+            <br>
+            Updated: 07/09/2024 09:05 AM
+            <div class="barChart_row" data-value="25.0">
+            <span class="barChart__value">25%</span>
+            <span class="barChart__bar" style=" background: rgb(239, 239, 239);">
+            <span class="barChart__barFill" style="background: #2A5B84;"></span>
+            </span>
+            </div>
+        </div>
+        <div class="barChart" style="width:90%;">
+            William Pitt Union
+            <br>
+            <span style="color:green">(Open)</span>
+            <br>
+            Last Count: 25
+            <br>
+            Updated: 07/09/2024 09:05 AM
+            <div class="barChart_row" data-value="25.0">
+            <span class="barChart__value">25%</span>
+            <span class="barChart__bar" style=" background: rgb(239, 239, 239);">
+            <span class="barChart__barFill" style="background: #2A5B84;"></span>
+            </span>
+            </div>
+        </div>
+        <div class="barChart" style="width:90%;">
+            Pitt Sports Dome
+            <br>
+            <span style="color:green">(Open)</span>
+            <br>
+            Last Count: 15
+            <br>
+            Updated: 07/09/2024 09:05 AM
+            <div class="barChart_row" data-value="20.0">
+            <span class="barChart__value">20%</span>
+            <span class="barChart__bar" style=" background: rgb(239, 239, 239);">
+            <span class="barChart__barFill" style="background: #2A5B84;"></span>
+            </span>
+            </div>
+        </div>
+        """

